blog/_posts/2020-02-12-local-dev.md

```diff
@@ -0,0 +1,86 @@
+---
+slug: local-dev
+date: 2020-02-11
+author: dan
+layout: blog
+title: "Non-Kubernetes Local Dev"
+subtitle: "Using Tilt to replace your start.sh"
+image: /assets/images/local-dev/uncontained.webp
+image_caption: "The goose from Untitled Goose Game feeling the freedom of not being in a container anymore."
+tags:
+  - tilt
+  - local
+keywords:
+  - tilt
+  - local
+---
+
+Many teams that use Kubernetes for production don't use Kubernetes in development to avoid Kubernetes pain points (like the slow inner-loops of building container images). But these teams still feel the pains of multi-service development:
+* Updating the right subset of microservices after you edit files is manual/error-prone/ad hoc even if you're just running bare processes on your laptop.
+* Finding a problem across many services? Playing 20 questions is as clumsy with `tail -f` or `ps` as `kubectl`.
+* Onboarding new users to this clusterf...un? You want them to be productive on day 1 without having to learn the whole stack, regardless of where it's running.
+
+Now you can use Tilt to solve these pains of multi-service development even if you aren't using Kubernetes for local development. Tilt can improve workflows built atop `foreman`, `docker-compose`, `edward`, a custom `start.sh, or a wiki page with tmux instructions.
+
+Expanding to support non-Kubernetes development environments gets us closer to our promise: if you use Kubernetes in Prod, you need Tilt for Dev.
```

Comment:
Talking about "promises" seems like a thing we would say internally but not something that makes a lot of sense externally.

Maybe I would rephrase it as: "Now if you use Kubernetes for prod, you should use Tilt for dev. Full stop."